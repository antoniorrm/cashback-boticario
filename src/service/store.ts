/* eslint-disable import/prefer-default-export */
import { store } from '../store'

export function getAccessToken() {
	return store.getState().auth.token
}