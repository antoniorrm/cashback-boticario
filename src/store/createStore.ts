import { createStore, applyMiddleware, Middleware, Reducer } from 'redux'
import { PersistState } from 'redux-persist'
import { AuthAction, AuthState } from './modules/auth/types'

export interface StoreState {
	_persist: PersistState
	auth: AuthState
}

export type StoreAction = AuthAction

export default (
	reducers: Reducer<StoreState, StoreAction>,
	middlewares: Middleware[]
) => {
	const enhancer = applyMiddleware(...middlewares)

	return createStore(reducers, enhancer)
}
