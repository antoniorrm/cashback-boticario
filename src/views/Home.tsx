import React from 'react'

const Home = () => <div>OIiii</div>

export default Home
