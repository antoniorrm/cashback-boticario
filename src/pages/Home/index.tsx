import React, { useEffect, useRef, useState } from 'react'
import {
	Box,
	Container,
	Fab,
	makeStyles,
	Tooltip,
	Typography,
} from '@material-ui/core'
import { Add } from '@material-ui/icons'
import { useSelector } from 'react-redux'

import PurchaseCard from './components/PurchaseCard'
import CashbackCard from '../../components/CashbackCard'
import PurchaseFormModal, { ModalHandler } from './components/PurchaseFormModal'
import api from '../../service/api'
import { StoreState } from '../../store/createStore'

const useStyles = makeStyles(theme => ({
	header: {
		[theme.breakpoints.up('sm')]: {
			display: 'flex',
		},
		alignItems: 'center',
		justifyContent: 'space-between',
		margin: '8px 0',
	},
	absolute: {
		position: 'fixed',
		bottom: theme.spacing(2),
		right: theme.spacing(3),
	},
}))

type Purchase = {
    id: number
	codigo: string
	value: number
	valuePercent: number
	percent: number
    date: string
	status: string
}

export default function Home() {
	const classes = useStyles()

	const refModal = useRef<ModalHandler>(null)

    const { userData } = useSelector((state: StoreState) => state.auth)

    const [purchases, setPurchases] = useState<Purchase[]>([])

    async function handlePurchases() {
        try {
            const response = await api.get(`/purchases?cpf=${userData?.cpf}`)
            setPurchases(response.data)
        } catch (error) {
            console.log(error)
        }
    }
    
    useEffect(() => {
		handlePurchases()
	}, [])

	return (
		<Container>
			<Box className={classes.header}>
				<Typography variant='h5' component='h2'>
					Bem vindo, {userData?.fullName}
				</Typography>
				<CashbackCard />
			</Box>
			<Typography variant='h5' component='h2'>
				Compras cadastradas
			</Typography>
			{purchases.map(purchase => (
				<PurchaseCard
					key={purchase.id}
					codigo={purchase.codigo}
					value={purchase.value}
					valuePercent={purchase.valuePercent}
					percent={purchase.percent}
					date={purchase.date}
					status={purchase.status}
					onEdit={() => {}}
					onDelete={() => {}}
				/>
			))}
            {purchases.length === 0 && <Typography color='textSecondary' align="center">Sem compras cadastrada!</Typography>}
			<Tooltip title='Adicionar compra' aria-label='add'>
				<Fab
					color='primary'
					className={classes.absolute}
					onClick={() => refModal.current?.handleOpen()}
				>
					<Add />
				</Fab>
			</Tooltip>

			<PurchaseFormModal ref={refModal} />
		</Container>
	)
}
