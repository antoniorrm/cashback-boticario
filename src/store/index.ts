import createSagaMiddleware from 'redux-saga'
import { AnyAction, Middleware, Store } from 'redux'
import { persistStore } from 'redux-persist'

import createStore from './createStore'
import rootReducer from './modules/rootReducer'
import rootSaga from './modules/rootSaga'
import persistReducers from './persistedReducers'

const sagaMiddleware = createSagaMiddleware()

const middlewares: Middleware[] = [sagaMiddleware]

const store = createStore(persistReducers(rootReducer), middlewares)
const persistor = persistStore(store as unknown as Store<any, AnyAction>)

sagaMiddleware.run(rootSaga)

export { store, persistor }
