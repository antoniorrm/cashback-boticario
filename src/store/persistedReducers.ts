import storage from 'redux-persist/lib/storage'
import { persistReducer } from 'redux-persist'
import { Reducer } from 'redux'
import { StoreAction, StoreState } from './createStore'

export default (reducers: Reducer<StoreState, StoreAction>) => {
	const persistedReducers = persistReducer(
		{
			key: 'cashback',
			storage,
			whitelist: ['auth'],
		},
		reducers
	)

	return persistedReducers
}
