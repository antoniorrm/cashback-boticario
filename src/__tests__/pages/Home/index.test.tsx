/* eslint-disable no-undef */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { SnackbarProvider } from 'notistack'
import { Provider } from 'react-redux'
import Home from '../../../pages/Home'
import { store } from '../../../store'

describe('Test Home page render correctly', () => {
	it('renders Home component correctly', () => {
		const { container } = render(
			<Provider store={store}>
				<SnackbarProvider maxSnack={1}>
					<Home />
				</SnackbarProvider>
			</Provider>
		)
		expect(container).toMatchSnapshot()
	})
	it('renders Texts in Home component correctly', () => {
		render(
			<Provider store={store}>
				<SnackbarProvider maxSnack={1}>
					<Home />
				</SnackbarProvider>
			</Provider>
		)
		const TextTitle = screen.getByText(/Bem vindo/i)
		const Text = screen.getByText(/Compras cadastradas/i)
		expect(TextTitle).toBeInTheDocument()
		expect(Text).toBeInTheDocument()
	})
})
