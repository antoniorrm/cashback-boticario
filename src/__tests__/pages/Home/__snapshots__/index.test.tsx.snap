// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home page render correctly renders Home component correctly 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="MuiBox-root MuiBox-root-3 makeStyles-header-1"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5"
      >
        Bem vindo, 
      </h2>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-4 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5"
          >
            Cashback acumulado
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            R$Â 0,00
          </p>
        </div>
      </div>
    </div>
    <h2
      class="MuiTypography-root MuiTypography-h5"
    >
      Compras cadastradas
    </h2>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      Sem compras cadastrada!
    </p>
    <button
      aria-label="add"
      class="MuiButtonBase-root MuiFab-root makeStyles-absolute-2 MuiFab-primary"
      tabindex="0"
      title="Adicionar compra"
      type="button"
    >
      <span
        class="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
